### ISSUE TITLE
Typescript: DateTimeOriginalis has wrong type of StringArrayTag

### ISSUE THREAD

**Issue by cybertino** [mattiasw/ExifReader#231]
### Description

Definition of `DateTimeOriginal` is not correct. It is declared as `StringArrayTag` but `value` is supposed to be a string instead of Array. Also there is `attributes` property instead of `id`.

**Additional details**

- ExifReader version: 4.11.1
- Node version: v16.17.0

Used this resource to confirm that this is supposed to be a string:

https://exiftool.org/TagNames/EXIF.html

**How to reproduce**

![jolla](https://user-images.githubusercontent.com/12428252/226498077-3c7a0fca-4092-43c4-9a34-1b28f273f8fc.jpg)

**Comment by mattiasw** [comment]
Hi!

This is a bit tricky, at least for me. I hadn't realize this problem was there. The reason for seeing this is because the combined output is used (in contrast to expanded): https://github.com/mattiasw/ExifReader#grouping
The XMP tag is overwriting the Exif tag. XMP tags all have the same type (with the `attribute` property and without `id`) while Exif tags have their individual ones. In the expanded output this works fine and `DateTimeOriginal` seems to be typed correctly in both separate cases. When they're combined, however, the type for all the tags together does not work correctly. I'm still learning TypeScript and I don't know how to do that correctly. It should be allowing `DateTimeOriginal` to be either of the Exif tag type or the generic XMP type but the intersection used (`Tags & XmpTags`) seems to only allow the property from the `Tags` interface. If you know how to make that work (without duplicating the whole `Tags` interface), please have a try.

**Comment by cybertino** [comment]
How about this:

```ts
export type Tags = {
    [K in keyof ExifTags]: ExifTags[K] | XmpTag | ValueTag;
}
```

I've updated my PR with that.

**Comment by mattiasw** [comment]
Merged and released as 4.12.0. Thanks for the effort! And I think the extended test file that was constructed together with this fix will be of much help in the future.

**Comment by cybertino** [comment]
Awesome! :)
